clients/unshaded/src/main/java/tachyon/client/RemoteBlockInStream.java

```diff
@@ -138,6 +142,23 @@
   }
 
   /**
+   * An alternative to construct a RemoteBlockInstream, bypassing all the checks.
+   * The returned RemoteBlockInStream can be used to call {#link #readRemoteByteBuffer}.
+   * TODO: remove this constructor, and modify all places using this constructor.
+   *
+   * @param file, any, could be null
+   * @param readType, any type
+   * @param blockIndex, any index
+   * @param ufsConf, any ufs configuration
+   * @param tachyonConf, any
+   * @param addFlag, add another field so that this constructor differentiates
+   */
+  public RemoteBlockInStream(TachyonFile file, ReadType readType, int blockIndex, Object ufsConf,
```

Comment:
Do you think there is a way to make this "fake" constructor private, and still expose the static method? Is there a way to make the static method use this private constructor, and proceed?